### PR TITLE
9 adding controllers

### PR DESCRIPTION
**Pull request by AnaLanda**

### CODE REVIEW

File: src/main/java/com/cinema/model/dto/CinemaHallResponseDto.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class CinemaHallResponseDto {
+    private long id;
```
Comment (jah701) [inline src/main/java/com/cinema/model/dto/CinemaHallResponseDto.java]: Memory save? :)

File: src/main/java/com/cinema/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -26,4 +26,9 @@ public User add(User user) {
     public Optional<User> findByEmail(String email) {
         return userDao.findByEmail(email);
     }
+
+    @Override
+    public Optional<User> findById(Long id) {
+        return userDao.findById(id);
```
Comment (jah701) [inline src/main/java/com/cinema/service/impl/UserServiceImpl.java]: Maybe it would be better if we return Optional only on dao layer?

**Review by jah701 — APPROVED** [review]
Very well done 👍

File: src/main/java/com/cinema/controllers/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    private final CinemaHallMapper cinemaHallMapper;
+
+    public CinemaHallController(CinemaHallService cinemaHallService,
+                                CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @PostMapping
+    public CinemaHall addCinemaHall(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controllers/CinemaHallController.java]: let's not return the model itself (check all controllers)

return either void or ResponseDto

File: src/main/java/com/cinema/controllers/OrderController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    public OrderController(OrderService orderService, OrderMapper orderMapper,
+                           UserService userService, ShoppingCartService shoppingCartService) {
+        this.orderService = orderService;
+        this.orderMapper = orderMapper;
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+    }
+
+    @PostMapping("/complete")
+    public Order completeOrder(@RequestParam String email) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controllers/OrderController.java]:
```suggestion
    public Order completeOrder(@RequestParam Long userId) {
```

File: src/main/java/com/cinema/controllers/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        this.shoppingCartService = shoppingCartService;
+        this.movieSessionService = sessionService;
+        this.userService = userService;
+        this.shoppingCartMapper = shoppingCartMapper;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSession(@RequestParam Long userId,
+                                        @RequestBody MovieSessionRequestDto dto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/controllers/ShoppingCartController.java]: let's just pass movieSessionId as the @RequestParam

File: src/main/java/com/cinema/dao/UserDao.java [hunk 6]
```diff
@@ -7,4 +7,6 @@
     User add(User user);
 
     Optional<User> findByEmail(String email);
+
+    Optional<User> findById(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/UserDao.java:11]: you call optional.get() in your controllers so many times

maybe let's return User here?

File: src/main/java/com/cinema/model/dto/CinemaHallResponseDto.java (hunk 1 above)
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/CinemaHallResponseDto.java]:
```suggestion
    private Long id;
```
the default value of long is 0L

File: src/main/java/com/cinema/model/dto/MovieSessionRequestDto.java [hunk 7]
```diff
@@ -4,6 +4,7 @@
 
 @Data
 public class MovieSessionRequestDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/MovieSessionRequestDto.java]: let's remove this)

File: src/main/java/com/cinema/model/dto/OrderRequestDto.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.model.dto;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class OrderRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/OrderRequestDto.java]: do you use this?

File: src/main/java/com/cinema/model/dto/TicketRequestDto.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class TicketRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/TicketRequestDto.java]: same

File: src/main/java/com/cinema/model/dto/UserResponseDto.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class UserResponseDto {
+    private Long id;
+    private String email;
+    private String password;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/UserResponseDto.java]: let's not show secure data

File: src/main/java/com/cinema/model/dto/TicketResponseDto.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class TicketResponseDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/TicketResponseDto.java]: I think this Dto (and the mapper as well) is redundant

File: src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java [hunk 12]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.model.dto;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<TicketResponseDto> tickets;
+    private String userEmail;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java]:
```suggestion
    private Long userId;
    private List<Long> ticketIds;
```

File: src/main/java/com/cinema/model/dto/OrderResponseDto.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class OrderResponseDto {
+    private Long id;
+    private List<TicketResponseDto> tickets;
+    private String orderDateTime;
+    private String userEmail;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/OrderResponseDto.java]: let's use userId and ticketIds here

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java [hunk 14]
```diff
@@ -0,0 +1,11 @@
+package com.cinema.model.dto;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ShoppingCartResponseDto {
+    private Long id;
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java]: these two have the same value, right?

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java (hunk 14 above)
Comment (AnaLanda) [reply src/main/java/com/cinema/model/dto/ShoppingCartResponseDto.java]: Yeah, they should be identical, you're right!

**Pull request merged**